Apps/W1/HybridGP/app/src/Migration/Support/PurchaseOrders/GPPOMigrator.codeunit.al

```diff
@@ -242,17 +294,127 @@ codeunit 40108 "GP PO Migrator"
+                PurchaseLine.SetRange("Document Type", PurchaseHeader."Document Type");
+                PurchaseLine.SetRange("Document No.", PurchaseHeader."No.");
+                PurchaseLine.SetFilter("Qty. to Receive", '>%1', 0);
+                if PurchaseLine.FindSet() then
+                    repeat
+                        if not ItemTrackingDictionary.ContainsKey(PurchaseLine."No.") then
+                            if Item.Get(PurchaseLine."No.") then
+                                if Item."Item Tracking Code" <> '' then begin
+                                    ItemTrackingDictionary.Add(Item."No.", Item."Item Tracking Code");
+                                    Item."Item Tracking Code" := '';
+                                    Item.Modify();
+                                end;
+                    until PurchaseLine.Next() = 0;
+
+                Codeunit.Run(Codeunit::"Purch.-Post", PurchaseHeader);
+            end;
 
-        if PurchaseHeader.Get(PurchaseHeader."Document Type"::Order, PurchaseLine."Document No.") then begin
-            PurchaseHeader.Validate(Status, PurchaseDocumentStatus::Released);
-            PurchaseHeader.Modify();
+        PostItemAdjustments();
 
-            PurchaseLine.Validate("Over-Receipt Code", GPCodeTxt);
-            PurchaseLine.Validate("Over-Receipt Quantity", OverReceiptQty);
-            PurchaseLine.Modify();
+        // Set the item tracking back
+        Clear(ItemTrackingCode);
+        foreach ItemNo in ItemTrackingDictionary.Keys do
+            if ItemTrackingDictionary.Get(ItemNo, ItemTrackingCode) then
+                if Item.Get(ItemNo) then begin
+                    Item."Item Tracking Code" := ItemTrackingCode;
+                    Item.Modify();
+                end;
+    end;
 
-            PurchaseHeader.Validate(Status, PurchaseDocumentStatus::Open);
-            PurchaseHeader.Modify();
+    local procedure PostItemAdjustments()
+    var
+        ItemJournalLine: Record "Item Journal Line";
+        ItemJournalBatch: Record "Item Journal Batch";
+    begin
+        ItemJournalLine.SetRange("Journal Batch Name", ItemJournalBatchNameTxt);
+        if ItemJournalLine.FindSet() then
+            Codeunit.Run(Codeunit::"Item Jnl.-Post Batch", ItemJournalLine);
+
+        Clear(ItemJournalLine);
+        ItemJournalLine.SetRange("Journal Batch Name", ItemJournalBatchNameTxt);
+        if not ItemJournalLine.IsEmpty() then
+            ItemJournalLine.DeleteAll();
+
+        ItemJournalBatch.SetRange("Name", ItemJournalBatchNameTxt);
+        ItemJournalBatch.DeleteAll();
+    end;
+
+    local procedure SetVendorDocumentNo(var PurchaseHeader: Record "Purchase Header")
+    var
+        GPPOPReceiptApply: Record GPPOPReceiptApply;
+        GPPOPReceiptHist: Record GPPOPReceiptHist;
+    begin
+        GPPOPReceiptApply.SetRange(PONUMBER, PurchaseHeader."No.");
+        GPPOPReceiptApply.SetFilter(Status, '%1', GPPOPReceiptApply.Status::Posted);
```

Comment:
SetRange is better if you are not using filter syntax

Reply:
Change made, thanks.